component/core/src/main/java/org/exoplatform/social/core/attachment/AttachmentServiceImpl.java

```diff
@@ -298,61 +322,91 @@ private List<String> getAttachmentFileIds(String objectType, String objectId) {
+  private ObjectAttachmentOperationReport attachUploadFiles(List<FileAttachmentObject> uploadedFiles,
                                                             String objectType,
                                                             String objectId,
                                                             String parentObjectId,
                                                             long userIdentityId) {
     ObjectAttachmentOperationReport report = new ObjectAttachmentOperationReport();
-    uploadIds.stream()
-             .distinct()
-             .map(uploadId -> {
-               UploadResource uploadedResource = uploadService.getUploadResource(uploadId);
-               if (uploadedResource == null) {
-                 LOG.warn("Uploaded resource with id " + uploadId + " wasn't found");
-                 report.addError(uploadId, "attachment.uploadIdNotFound");
-               }
-               return uploadedResource;
-             })
-             .filter(Objects::nonNull)
-             .forEach(uploadResource -> createAttachment(uploadResource,
-                                                         objectType,
-                                                         objectId,
-                                                         parentObjectId,
-                                                         userIdentityId,
-                                                         report));
+    uploadedFiles.stream().distinct().map(uploadedFile -> {
+      UploadedAttachmentDetail uploadedAttachmentDetail = new UploadedAttachmentDetail();
+      UploadResource uploadedResource = uploadService.getUploadResource(uploadedFile.getUploadId());
+      if (uploadedResource == null) {
+        LOG.warn("Uploaded resource with id " + uploadedFile.getUploadId() + " wasn't found");
+        report.addError(uploadedFile.getUploadId(), "attachment.uploadIdNotFound");
+      }
+      uploadedAttachmentDetail.setId(uploadedFile.getId());
+      uploadedAttachmentDetail.setAltText(uploadedFile.getAltText());
+      uploadedAttachmentDetail.setUploadedResource(uploadedResource);
+      return uploadedAttachmentDetail;
+    })
+                 .filter(Objects::nonNull)
+                 .forEach(uploadedAttachmentDetail -> createAttachment(uploadedAttachmentDetail,
+                                                                       objectType,
+                                                                       objectId,
+                                                                       parentObjectId,
+                                                                       userIdentityId,
+                                                                       report));
     return report;
   }
 
-  private void createAttachment(UploadResource uploadResource,
+  private void createAttachment(UploadedAttachmentDetail uploadedAttachmentDetail,
```

Comment:
please split this methods into 3 methods:

* saveAttachment, that calls
* either createAttachment
* or updateAttachment

Reply:
done